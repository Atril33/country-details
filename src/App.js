import { useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { Route, Routes } from 'react-router-dom';
import { countryDataFetch } from './Redux/country/countrySlice';
import Home from './components/Home/Home';
import InfoPage from './components/InfoPage/InfoPage';

function App() {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(countryDataFetch());
  });
  
  return (
    <Routes>
      <Route path="/" element={<Home />} />
      <Route path="/country/:name" element={<InfoPage />} />
    </Routes>
  );
}

export default App;
