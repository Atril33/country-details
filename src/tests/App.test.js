import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import App from '../App';
import store from '../Redux/store';

describe('App component', () => {
  test('Should match snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
