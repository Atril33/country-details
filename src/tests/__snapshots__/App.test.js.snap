// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Should match snapshot 1`] = `
<DocumentFragment>
  <header>
    <div
      class="top-bar"
    >
      <img
        alt="Home Icon"
        class="home-img"
        src="Icon-home.png"
      />
      <h1>
        Countries
      </h1>
      <div
        class="right-side"
      >
        <img
          alt="Microphone Icon"
          src="icon-microphone.png"
        />
        <img
          alt="Setting Icon"
          src="icon-setting.png"
        />
      </div>
    </div>
    <img
      alt="World Map"
      class="bannar-image"
      src="country-map.png"
    />
    <div
      class="input-cont"
    >
      <input
        class="search-input"
        placeholder="🔍 Search Country"
        type="text"
        value=""
      />
    </div>
  </header>
  <main>
    <div
      class="status-bar"
    >
      <h2
        class="status-txt"
      >
        Status by country
      </h2>
    </div>
    <div
      class="country-container"
    >
      ;
    </div>
  </main>
</DocumentFragment>
`;
